File: DynamicKey/AgoraDynamicKey/java/src/main/java/io/agora/sample/ApaasTokenBuilderSample.java
package io.agora.sample;

import io.agora.apaas.ApaasTokenBuilder;

public class ApaasTokenBuilderSample {
    // Need to set environment variable AGORA_APP_ID
    private static String appId = System.getenv("AGORA_APP_ID");
    // Need to set environment variable AGORA_APP_CERTIFICATE
    private static String appCertificate = System.getenv("AGORA_APP_CERTIFICATE");

    private static String roomUuid = "123";
    private static String userUuid = "2882341273";
    private static Short role = 1;
    private static int expire = 600;

    public static void main(String[] args) {
        System.out.printf("App Id: %s\n", appId);
        System.out.printf("App Certificate: %s\n", appCertificate);
        if (appId == null || appId.isEmpty() || appCertificate == null || appCertificate.isEmpty()) {
            System.out.printf("Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE\n");
            return;
        }

        ApaasTokenBuilder tokenBuilder = new ApaasTokenBuilder();
        String roomUserToken = tokenBuilder.buildRoomUserToken(appId, appCertificate, roomUuid, userUuid, role, expire);
        System.out.printf("Apaas room user token: %s\n", roomUserToken);

        String userToken = tokenBuilder.buildUserToken(appId, appCertificate, userUuid, expire);
        System.out.printf("Apaas user token: %s\n", userToken);

        String appToken = tokenBuilder.buildAppToken(appId, appCertificate, expire);
        System.out.printf("Apaas app token: %s\n", appToken);
    }
}

File: DynamicKey/AgoraDynamicKey/kotlin/src/sample/kotlin/io/agora/sample/ApaasTokenBuilderSample.kt
package io.agora.sample

import io.agora.media.ApaasTokenBuilder

/** Demonstrates APaaS Token007 generation. */
object ApaasTokenBuilderSample {
    /** Generates and prints APaaS room-user, user, and application sample tokens. */
    @JvmStatic
    fun main(args: Array<String>) {
        val appId = System.getenv("AGORA_APP_ID")
        val appCertificate = System.getenv("AGORA_APP_CERTIFICATE")
        if (appId.isNullOrEmpty() || appCertificate.isNullOrEmpty()) {
            println("Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE")
            return
        }

        val builder = ApaasTokenBuilder()
        println(
            "APaaS room-user token: ${builder.buildRoomUserToken(appId, appCertificate, ROOM_UUID, USER_UUID, ROLE, EXPIRE)}"
        )
        println("APaaS user token: ${builder.buildUserToken(appId, appCertificate, USER_UUID, EXPIRE)}")
        println("APaaS app token: ${builder.buildAppToken(appId, appCertificate, EXPIRE)}")
    }

    private const val ROOM_UUID = "123"
    private const val USER_UUID = "2882341273"
    private const val ROLE: Short = 1
    private const val EXPIRE = 600
}
